core/dbt/parser/manifest.py

```diff
@@ -270,20 +274,17 @@ def load(self):
             # We need to parse the macros first, so they're resolvable when
             # the other files are loaded
             start_load_macros = time.perf_counter()
-            for project in self.all_projects.values():
-                if project.project_name not in project_parser_files:
-                    continue
-                parser_files = project_parser_files[project.project_name]
-                if 'MacroParser' not in parser_files:
-                    continue
-                parser = MacroParser(project, self.manifest)
-                for file_id in parser_files['MacroParser']:
-                    block = FileBlock(self.manifest.files[file_id])
-                    parser.parse_file(block)
-                    # increment parsed path count for performance tracking
-                    self._perf_info.parsed_path_count = self._perf_info.parsed_path_count + 1
-            # Look at changed macros and update the macro.depends_on.macros
-            self.macro_depends_on()
+            self.load_and_parse_macros(project_parser_files)
+
+            # If we're partially parsing check that certain macros have not been changed
+            if self.partially_parsing and self.skip_partial_parsing_because_of_macros():
```

Comment:
We're checking if we should skip partial parsing here, but we also check above. Can we merge them into one check? I like the interface above where we just have one clear check.

Reply:
No, we can't merge them into one place. The new place to skip requires that the macros be loaded, which they aren't in the first place. Yes, this makes the code more complicated but there's no way around it if we want to do this.

Reply:
Okay figured I'd check.